src/Router.cpp

```diff
@@ -12,13 +12,29 @@ Router::~Router()
 HTTPResponse Router::routeRequest(const HTTPRequest &request)
 {
 	HTTPResponse response;
```

Comment:
A response is created also in the Router.

Reply:
I am not sure if we are creating an object, I think at this level the model might be wrong:
https://chat.openai.com/share/2b9c431e-827a-48d5-821f-0395b7f2b022

But it says it doesn't matter, so it seems to be just a matter of what is aesthetic and easy to read, what do you think ?

Reply:
But I agree it is better to create or declared the object only once, I will modify the function, except maybe those from CGIHandler

Reply:
I was more saying that we declare HTTPResponse like `HTTPResponse response` in many places in the code and actually we should create just 1 HTTPResponse object for each Connection. But if it works probably at the moment it doesn't matter. They will be just extra response object that get just destroyed or not created.

Reply:
Yes that's what I understood,  they might be " temporary objects". Anyway I think you are right and I got rid of all of them I believe. it seems a bit harder to read but it is probably because of C vs C++ and it seems to be good practice